lib/webpage_cache_service/render.rb

```diff
@@ -38,19 +38,29 @@ def nokogiri
 
     def rewrite_links!
       ASSET_XPATHS.each do |xpath|
+        nokogiri.xpath(xpath).each do |xpath_attr|
+          xpath_attr.value = rewrite_asset_link xpath_attr.value
+        end
+      end
+
+      LINK_XPATHS.each do |xpath|
         nokogiri.xpath(xpath).each do |xpath_attr|
           xpath_attr.value = rewrite_link xpath_attr.value
         end
       end
     end
 
-    def rewrite_link url
+    def rewrite_asset_link url
       link = uri + Addressable::URI.parse( url )
       link_key = Digest::SHA256.hexdigest link.to_s
 
       base_uri + link_key
     end
 
+    def rewrite_link url
+      uri + Addressable::URI.parse( url )
```

Comment:
fancy!

Reply:
Addressable makes this nice because it'll turn it into a full link if the url with `uri` as the base, if its relative

Reply:
names are hard